### DISCUSSION TITLE
How to use 'graph. eval()'

### DISCUSSION THREAD

**Discussion by LiJie20190102** [JanusGraph/janusgraph#4147] [Q&A]
I use 'graph. eval()' to execute the gremlin script, if 'g. V()', what I expected was that this would return all vertices, but it definitely looks like 'g. V()' has obtained an object that I am not clear about. Should I handle this object?

![image](https://github.com/JanusGraph/janusgraph/assets/53458004/a32ae760-db4d-43cd-866a-0093461b8f0e)

**Comment by LiJie20190102** [comment]
@li-boxuan please help me

**Reply by li-boxuan** [reply]
@LiJie20190102 Try `g.V().toList()`

**Reply by LiJie20190102** [reply]
If I use 'Dynamic Graphs', how should I use' gremlin remote 'to execute gremlin scripts. I see that 'a WebSocket or HTTP Endpoint' requires a static graph before it can be used, Do you have any relevant information？

**Reply by li-boxuan** [reply]
What's your deployment scenario? Are you using JanusGraph as an embedded library?

**Reply by LiJie20190102** [reply]
yeah

**Reply by li-boxuan** [reply]
Then you could do something in your java app like this:

```java
final StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.open(graphName);
graph.eval(script);
```

**Reply by LiJie20190102** [reply]
OK, I understand this form now.
If I want to remotely use HTTP to call the gremlin script while using dynamic graphs, what should I do?

**Reply by li-boxuan** [reply]
> If I want to remotely use HTTP to call the gremlin script while using dynamic graphs

I am not sure what kind of scenario that is. I assume you want to add a REST API to your application that takes a gremlin script as input. If this is the case, you could let your REST API take one more parameter: graph name.

**Reply by LiJie20190102** [reply]
this scenario :
![image](https://github.com/JanusGraph/janusgraph/assets/53458004/4934c779-1449-44af-bf59-05c60544e780)
![image](https://github.com/JanusGraph/janusgraph/assets/53458004/2e35eb0c-0678-47d0-9f65-5f58132b4daa)

**Reply by li-boxuan** [reply]
You are using JanusGraph server, not JanusGraph as an embedded library.

**Reply by LiJie20190102** [reply]
Sorry, if I use 'JanusGraph server', how should I use the REST API method mentioned above

**Reply by LiJie20190102** [reply]
Do you have any good suggestions？
